p2p/libp2p/networksharding/listsSharder.go

```diff
@@ -148,16 +156,19 @@ func (ls *listsSharder) ComputeEvictionList(pidList []peer.ID) []peer.ID {
 	existingNumIntraShardObservers := len(peerDistances[intraShardObservers])
 	existingNumCrossShardValidators := len(peerDistances[crossShardValidators])
 	existingNumCrossShardObservers := len(peerDistances[crossShardObservers])
+	existingNumFullHistoryObservers := len(peerDistances[fullHistoryObservers])
 	existingNumUnknown := len(peerDistances[unknown])
 
 	var numIntraShardValidators, numCrossShardValidators int
 	var numIntraShardObservers, numCrossShardObservers int
+	var numFullHistoryObservers int
 	var numUnknown, remaining int
 
 	numIntraShardValidators, remaining = computeUsedAndSpare(existingNumIntraShardValidators, ls.maxIntraShardValidators)
 	numCrossShardValidators, remaining = computeUsedAndSpare(existingNumCrossShardValidators, ls.maxCrossShardValidators+remaining)
 	numIntraShardObservers, remaining = computeUsedAndSpare(existingNumIntraShardObservers, ls.maxIntraShardObservers+remaining)
 	numCrossShardObservers, remaining = computeUsedAndSpare(existingNumCrossShardObservers, ls.maxCrossShardObservers+remaining)
+	numFullHistoryObservers, remaining = computeUsedAndSpare(existingNumFullHistoryObservers, ls.maxFullHistoryObservers+remaining)
```

Comment:
I do not think it is wise to call this function adding the remaining. As discussed, it should be more like:
```GO
numFullHistoryObservers, _ = computeUsedAndSpare(existingNumFullHistoryObservers, ls.maxFullHistoryObservers)
```
The logic behind this: keep the full history nodes list completely decoupled of the "cascading remaining" applied on all other lists. Not even to unknown list, it has plenty slots to populate from the other lists.

Reply:
Done